### ISSUE TITLE
Bug: Crash on macOS when attempting to add account

### ISSUE THREAD

**Issue by thattridentdude** [Dimillian/IceCubesApp#1069]
## Environment:

 - OS: 
 macOS 13.3
 - IceCubesApp version: 
1.5.9 / current main branch (2ba2675ae443c7a61e091684163d39c9d13160fe) 

## Description

When starting the app for the first time on macOS you're offered to add an account. 
Attempting to do this by trying to add a mastodon.online account crashes the app the moment it tries to hand off login to Safari. 

I'm currently still investigating and will provide PR if I can find cause and fix, but I've found out the following so far:

That said, this might be a separate issue because manually typing `mastodon.online` in full doesn't prevent the crash. 

Attached is a zip with console output (incl. stacktrace) and unsymbolicated crash reports from App Store version. 

## Related Issues

 - [X] Search that this bugs don't already exist before creating it.
 
[IceCubesCrashData.zip](https://github.com/Dimillian/IceCubesApp/files/10832373/IceCubesCrashData.zip)

-----

Edit: 

Removed console lines about incomplete url, is unrelated to this issue. 

Issue is caused by `SFSafariViewController` being broken on the macOS 13.3 dev beta, including the most recent one (22E5230e). 

13.3 [release notes](https://developer.apple.com/documentation/macos-release-notes/macos-13_3-release-notes) make no mention of any changes that could affect this, so this might be a bug. 

Nevertheless, I've created a workaround in #1117 and will file a FB with Apple. 

If this is new (intended) behavior or workaround is necessary long term for whatever reason, then I suggest polishing it up the workaround a bit.

Will reply with comment once/if I have more information about FB report to Apple.

**Comment by thattridentdude** [comment]
Reported to Apple under FB12021800. 

Tried to create minimal reproducible example (see attached) but I'm unable to reproduce the issue in the example. 

I think I'm going to need help from someone who is more familiar with the codebase to see if we can add in some more elements from Ice Cube App and see if that might reproduce the issue. 


[macOS 13.3 SFSafariViewController Bug.zip](https://github.com/Dimillian/IceCubesApp/files/10856155/macOS.13.3.SFSafariViewController.Bug.zip)

**Comment by Dimillian** [comment]
So PhotoPicker is broken in the same sort of way on iOS 16.4. Really hope it's just a in flight beta issue and will not ship like that. Will probably wait for RC before I upgrade myself. If at this point it's still like this, then we'll go with your fix.

**Comment by thattridentdude** [comment]
Yeah, I saw your post on Mastodon earlier today about the PhotoPicker. Not sure what's going on over at Apple.

I've changed the string in the merge request so it's ready to go for whenever you want the fix. I've tested it on iOS/iPadOS and stable macOS and those aren't affected. 

But you might want to consider adding some polish to it before you merge it into main. 

The screen looks a bit bland, the there's no sheet animation and dismissing it with the `Cancel` in the upper left dismisses both sheets because they're bound to `@Environment(\.dismiss) private var dismiss`.
Minor gripes that only a small subset of users will experience but it depends on how important it is to you.


![CleanShot 2023-02-28 at 22 41 36](https://user-images.githubusercontent.com/5077094/222063954-19298afa-3992-4a81-96da-45af46dc3b8f.png)
